### ISSUE TITLE
Fix sorting

### ISSUE THREAD

**Issue by chilagrow** [FerretDB/FerretDB#1637]
### Versions

0.7.0

### What did you do?

Added compat sort test `TestQueryCompatSort`

https://github.com/FerretDB/FerretDB/blob/5920ba1b88ac40135c40b7b2024ae66b70bbeecc/integration/query_compat_test.go#L153

### What did you expect to see?

The sorting returns the same as compat.

### What did you see instead?

The sort order for array for composite is not the same. It's because array sort order should check the elements within the array for sorting. https://www.mongodb.com/docs/manual/reference/bson-type-comparison-order/#arrays

https://github.com/FerretDB/FerretDB/blob/05950cb4ffb0604e96d51312747faad6afac143c/integration/query_compat_test.go#L154

```
        --- FAIL: TestQueryCompatSort/Asc/TestQueryCompatSort_Composites (0.01s)
            query_compat_test.go:168: Compat (expected) IDs: [array-empty array-composite array-null array-three array-three-reverse array array-numbers-asc array-two array-strings-desc document-empty document-null array-documents document document-composite document-composite-reverse]
            query_compat_test.go:168: Target (actual)   IDs: [document-empty document-null document document-composite document-composite-reverse array-empty array-null array-three-reverse array array-numbers-asc array-three array-two array-composite array-strings-desc array-documents]
            query_compat_test.go:168: 

```

https://github.com/FerretDB/FerretDB/blob/05950cb4ffb0604e96d51312747faad6afac143c/integration/update_field_compat_test.go#L121
```
        	            	--- expected
        	            	+++ actual
        	            	@@ -3,8 +3,8 @@
        	            	     "_id",
        	            	-    "a",
        	            	-    "v"
        	            	+    "v",
        	            	+    "a"
        	            	   ],
        	            	   "_id": "unset",
        	            	-  "a": 30,
        	            	-  "v": 39
        	            	+  "v": 39,
        	            	+  "a": 30
        	            	 }
```

**Comment by chilagrow** [comment]
Part of this is fixed in #1634 and remaining maybe fixed in #1652.

**Comment by chilagrow** [comment]
Fixed by #1652
